File: iosSDK/text/Block/Block/ViewController.m
//
//  ViewController.m
//  Block
//
//  Created by  wj on 16/5/3.
//  Copyright © 2016年 gzq. All rights reserved.
//

#import "ViewController.h"
#import "RootViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"页面1";
    self.view.backgroundColor = [UIColor whiteColor];
    
    void (^myBlock)(NSString *str);
    myBlock = ^(NSString *str){
        NSLog(@"%@",str);
    };
    myBlock(@"1111");
    // Do any additional setup after loading the view, typically from a nib.
}
-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    RootViewController *root = [[RootViewController alloc] init];
    
    root.myBlock = ^(UIColor *color,NSString *name){
        self.view.backgroundColor = color;
        self.title = name;
    };
    [self.navigationController pushViewController:root animated:YES];
    
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: oop/weatherData/weatherData/ViewController.swift
//
//  ViewController.swift
//  weatherData
//
//  Created by  wj on 15/10/31.
//  Copyright © 2015年 gzq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let weatherData = WeatherData()
        var currentDisplay = CurrentConditionsDisplay(weatherData: weatherData)
        weatherData.setMeasurements(80, humidity: 65, pressure: 30.4)
        weatherData.setMeasurements(82, humidity: 70, pressure: 29.2)

        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

